tests/llm/utils/mock_dal.py

```diff
@@ -0,0 +1,78 @@
+import json
+import logging
+from pathlib import Path
+from typing import Dict, Optional
+
+from pydantic import TypeAdapter
+
+from holmes.core.supabase_dal import SupabaseDal
+from holmes.core.tool_calling_llm import ResourceInstructions
+from tests.llm.utils.constants import AUTO_GENERATED_FILE_SUFFIX
+from tests.llm.utils.mock_utils import read_file
+
+class MockSupabaseDal(SupabaseDal):
+
+    def __init__(self, test_case_folder:Path, issue_data:Optional[Dict], resource_instructions:Optional[ResourceInstructions], dal_passthrough:bool):
+        super().__init__()
+        self._issue_data = issue_data
+        self._resource_instructions = resource_instructions
+        self._test_case_folder = test_case_folder
+        self._dal_passthrough = dal_passthrough
+
+    def get_issue_data(self, issue_id: Optional[str]) -> Optional[Dict]:
+        if self._issue_data is not None:
+            return self._issue_data
+        else:
+            data = super().get_issue_data(issue_id)
+            file_path = self._get_mock_file_path("issue_data")
+
+            with open(file_path, 'w') as f:
+                f.write(json.dumps(data or {}, indent=2))
+                f.close()
+
+            logging.warning(f"A mock file was generated for you at {file_path} with the contentof dal.get_issue_data({issue_id})")
+            if self._dal_passthrough:
```

Comment:
Just to make sure i understand, that passthourgh check is here after the call so we will be able to autogenerate the data? 
Maybe we should have dedicated task for generating relevant data instead so it will simplify the code itself?

Reply:
> that passthourgh check is here after the call so we will be able to autogenerate the data?
Yes

>we should have dedicated task for generating relevant data instead so it will simplify the code itself?
Yes, I agree. That also links with the current ai-tests methodology that relies on a cluster that has these issues. We should either fall back on "live" behaviour with the ability to generate mocks, or have simple mocks with simpler logic.